### DISCUSSION TITLE
moving commands from `rust` to the `std` library

### DISCUSSION THREAD

**Discussion by amtoine** [nushell/nushell#9061] [Ideas]
### [Go back to the roadmap.](https://github.com/nushell/nushell/issues/8450)

i've been wanting to discuss this for while, so here we are :yum: 

one of the goal of the library is to relieve the main source base of `rust` commands by moving non-time-sensitive commands to the `nu-std` crate :+1: 

in https://github.com/nushell/nushell/issues/8311, we already have some ideas
- the `help` commands
- the "shell" commands
- the benchmark command
- the banner

now comes the question...
## what could / should we move to the standard library? :smirk: 

currently the following commands are in the process of being moved or have been:
:point_right: from the discord server:
- [ ] "Re-implement `enter`, `shells`, `n` and `p` commands" (see [#standard-library](https://discord.com/channels/601130461678272522/1075541668922658868/1079666283735429161))
  - #8368 
  - #8415 
- [ ] a `benchmark` to wrap `timeit` in a multi-run loop ( see [#cool-scripts](https://discord.com/channels/601130461678272522/615253963645911060/1081872279178334318))
  - https://github.com/nushell/nushell/issues/8696

:point_right: from *GitHub*
- [ ] move the `help` subcommands from `rust` to `nushell` and keep the main `help` in the `rust` implementation to still allow `help -f ...` everywhere.
  - [`print_help.nu` from @jntrnr](https://gist.github.com/jntrnr/bb7f8827db14131ab95229e1c1efcc42)
  - https://github.com/nushell/nushell/pull/8505
  - https://github.com/nushell/nushell/pull/8611
  - https://github.com/nushell/nushell/issues/8813
  - https://github.com/nushell/nushell/pull/8983: there are great ideas in this PR but due to the lack of contributor we closed it for now :relieved: 

> **Note**
> comes from a comment from @kubouch in #8403 
> > Ultimately, the `help` commands could be re-implemented in Nu code as a part of the standard library so that's also something to consider in the future.

**Comment by fdncred** [comment]
We need to be careful with moving commands. We already know we don't want to move performance-sensitive commands to the standard library. But we also need to keep in mind that everything we add to the standard library slows down nushell startup speed. Right now, on my mac it's about a 115ms slower. On Windows, I'm sure it's dramatically slower.

I think we just need to be conscious of the impact that it's having until we can speed up the startup process/parser. It may also mean we learn to write tighter nushell script code that isn't as verbose but yields the same results?

I'd also want to get done what we have queued up before moving to other things. I'd like to see it completed, and live with it for a while to see how we like it before moving on.

**Reply by amtoine** [reply]
absolutely agree :ok_hand: 
just wanted to have that opened :yum:
